src/utils/extract_tables.py

```diff
@@ -0,0 +1,170 @@
+"""
+Simple table extraction module for PDF parsing.
+Can be imported into PdfParser class to add table extraction capabilities.
+"""
+
+import camelot
+import pandas as pd
+import json
+from typing import List, Dict, Any
+
+
+def extract_tables_from_pdf(pdf_path: str, pages: str = "all") -> List[Dict[str, Any]]:
+    """
+    Extract tables from PDF and return them as structured data.
+
+    Args:
+        pdf_path (str): Path to the PDF file
+        pages (str): Page numbers to extract from (default: 'all')
+
+    Returns:
+        List[Dict]: List of dictionaries containing table data and metadata
+    """
+    try:
+        tables = camelot.read_pdf(
+            pdf_path, pages=pages, flavor="lattice", strip_text="\n"
+        )
+
+        extracted_tables = []
+
+        for i, table in enumerate(tables):
+            parsing_report = table.parsing_report
+            accuracy = parsing_report.get("accuracy", 0)
+
+            df = clean_table_dataframe(table.df)
+
+            if df.empty or df.shape[0] < 2 or df.shape[1] < 2:
+                continue
+
+            content_ratio = calculate_content_ratio(df)
+
+            if content_ratio > 0.1 or accuracy > 40:
+                table_data = {
+                    "table_id": i,
+                    "page": table.parsing_report.get("page", "unknown"),
+                    "accuracy": round(accuracy, 2),
+                    "content_ratio": round(content_ratio, 2),
+                    "shape": df.shape,
+                    "data": df.to_dict("records"),
+                    "json": df.to_json(orient="records", force_ascii=False),
+                }
+                extracted_tables.append(table_data)
+
+        return extracted_tables
+
+    except Exception as e:
+        print(f"Error extracting tables from {pdf_path}: {e}")
+        return []
+
+
+def clean_table_dataframe(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Clean a DataFrame by removing empty rows/columns and whitespace.
+
+    Args:
+        df (pd.DataFrame): Raw DataFrame from table extraction
+
+    Returns:
+        pd.DataFrame: Cleaned DataFrame
+    """
+
+    cleaned_df = df.copy()
+
+    cleaned_df = cleaned_df.dropna(how="all", axis=0)
```

Comment:
```suggestion
    cleaned_df.dropna(how="all", axis=0, inplace=True)
```
That way you don't create a copy, so you are saving space (RAM).